craft_parts/parts.py

```diff
@@ -47,40 +47,431 @@
+    )
+    """If the source is a Git repository, this key specifies how far back in the
+    commit history to fetch.
+
+    During the pull step, the part fetches the repository from the specified commit
+    up to the target commit, the target tag, or the tip of the target branch.
+
+    Equivalent to the
+    :literalref:`--depth<https://git-scm.com/docs/git-pull#Documentation/git-pull.txt---depthdepth>`
+    parameter of ``git fetch``.
+
+    If unset, the part fetches the full repository history up to the target commit, the
+    target tag, or the tip of the target branch.
+    """
+
+    source_subdir: str = Field(
+        default="",
+        description="The subdirectory of the unpacked source where the build will occur.",
+        examples=["src", "demo_nodes_cpp"]
+    )
+    """The subdirectory of the unpacked source where the build will occur.
+
+    During the pull step, the build will be restricted to the specified path.
+
+    If unset, the build can access the entire file tree of the source.
+    """
+
+    source_submodules: list[str] | None = Field(
+        default=None,
+        description="The registered submodules to fetch from Git sources.",
+        examples=["[third_party/googletest, third_party/jsoncpp]", "[libbpf]"],
+    )
+    """If the source is a Git repository, this key specifies the registered Git
+    submodules that the project also needs.
+
+    During the pull step, the part fetches these submodules.
+
+    Equivalent to the
+    :literalref:`--recurse-submodules<https://git-scm.com/docs/git-clone#Documentation/git-clone.txt-code--recurse-submodulesltpathspecgtcode>`
+    parameter of ``git clone``.
+
+    If unset, the part will fetch all of the repository's submodules.
+    """
+
+    source_tag: str = Field(
+        default="",
+        description="The target tag for Git sources.",
+        examples=["1.0.1"],
+    )
+    """If the source is a Git repository, this key specifies the target tag.
+
+    During the pull step, the part fetches the repository from the earliest available
+    commit up to the commit with this tag.
+
+    The commit history can be truncated by specifying a ``source-depth``.
+
+    This key is mutually incompatible with ``source-branch`` and ``source-commit``.
+    """
+
+    source_type: str = Field(
+        default="",
+        description="The format of the part's source.",
+        examples=["git", "local"],
+    )
+    """The format of the part's source.
+
+    During the pull step, the part expects the source to behave like the specified
+    format.
+
+    If unset, the part attempts to auto-detect the format.
+
+    Supported formats include container types like ``.tar`` files and Debian packages,
+    version-controlled directories like Git repositories, and local files.
+
+    **Values**
+
+    .. list-table::
+        :header-rows: 1
+
+        * - Value
+          - Description
+        * - ``deb``
+          - Debian package
+        * - ``file``
```

Comment:
I believe this should be ``local``, not `file`.

Reply:
Can you add a second entry? It should be:

- local: A local directory or file within the project's directory.
- file: A "simple" or "plain" file to retrieve from the internet.